pkg/external_repos/introspect.go

```diff
@@ -49,6 +50,10 @@ func Introspect(repo *dao.Repository, repoDao dao.RepositoryDao, rpm dao.RpmDao)
 	)
 	log.Debug().Msg("Introspecting " + repo.URL)
 
+	if repo.FailedIntrospectionsCount > FailedIntrospectionsLimit && !repo.Public {
+		return 0, fmt.Errorf("introspection skipped because this repository has failed more than %v times in a row", FailedIntrospectionsLimit)
+	}
```

Comment:
During the nightly introspection, I /think/ these shouldn't necessarily be treated as errors, but instead should just be ignored.   That said, if introspect() is explicitly called on it, we do probably want to error i think?  So my suggestion would be to update the query that fetches the list of repos for introspection to filter these out automatically.

Reply:
I think my idea here was I wanted the UI to show "this repository is failing too many times so introspection has been stopped". But I guess I can communicate that by checking the count first, if that makes sense?